### PR TITLE
vboot: presentation

### PR DESCRIPTION
**Pull request by mkopec**

### CODE REVIEW

File: verified_boot.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+---
+
+# CBFS Verification
+
+- A relatively new coreboot feature
+- Cryptographically verifies components of the coreboot image
+- According to documentation:
+  > This only makes sense if you use some
+  > out-of-band mechanism to guarantee the integrity of the bootblock
+  > itself, such as Intel Boot Guard or flash write-protection.
```
Comment (krystian-hebel) [inline verified_boot.md:174]: Probably worth mentioning that this is the same as with Vboot.

File: verified_boot.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+  - A and B slots for updates, RO slot contains IBB and verification code
+  - Anti-rollback using TPM
+  - Vboot aware OS service manages slot updates
+- Dasharo:
+  - Initial idea was to leave RO untouched and only update A/B slots
+  - In practice, most updates introduce breaking changes that require updating
+    the bootblock
+  - Dasharo Tools Suite handles updates, does the entire update in one operation
+    - So we don't use the A/B feature
+  - Capsule Updates will help here
```
Comment (krystian-hebel) [inline verified_boot.md:196]: `Will` is too strong word, this is something that we may end up skipping.

```suggestion
  - Capsule Updates may help here
```

File: verified_boot.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+---
+
+# Other firmware update mechanisms
+
+- Flashrom plugin in fwupd
+  - Updates the BIOS region only
+  - Is not Vboot aware
+- UEFI Capsule Update
+  - Most widely used in proprietary UEFIs
+  - Requires an update handler in SMM
```
Comment (krystian-hebel) [inline verified_boot.md:207]: This is more of a chipset feature than requirement of Capsule Update.

File: verified_boot.md [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+- Intel BIOS Guard
+  - Replaces the SMM capsule update handler with an ACM
+  - Proprietary tooling
+  - Can bypass chipset flash protections
+  - Can also handle EC updates
+  - Proprietary feature with proprietary tooling
+- Top Swap
+  - Redundant bootblock feature
+  - Configurable up to 4M
```
Comment (krystian-hebel) [inline verified_boot.md:227]: Does this size depend on CPU family? I think I've seen somewhere that it is always 128K, but maybe I'm mistaken.

File: verified_boot.md [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+---
+
+# Verified boot
+
+- Verified Boot: Cryptographic verification of the boot process to ensure only
+  code from a trusted source (e.g. device vendor) can run
+- coreboot supports Vboot, Google's version of verified boot scheme
+- Vboot has:
+  - A root of trust
+  - Firmware verification and measurements to TPM
```
Comment (miczyg1) [inline verified_boot.md:47]: vboot has been decoupled from measured boot.

File: verified_boot.md [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+
+- Cr50 is a Google's TPM-like device, also known as Titan C
+
+---
+
+# Verified Boot in Dasharo
+
+.center.image-20[![](/remark-templates/dasharo-presentation-template/images/dasharo-sygnet.svg)]
+
+- Write protection is provided by chipset
```
Comment (miczyg1) [inline verified_boot.md:87]: can be also provided by SPI chip, but it is HW dependent to have a WP pin exposed. Also SPI chip protection has restrictions on the regions that can be protected

File: verified_boot.md [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+
+???
+
+- Capsule Updates are a requirement for Windows sticker certification
+- Supported by Windows Update and fwupd
+
+# Other firmware update mechanisms
+
+- Intel BIOS Guard
+  - Replaces the SMM capsule update handler with an ACM
```
Comment (miczyg1) [inline verified_boot.md:220]: It doesn't replace anything. Instead the SMM handler responsible for flash writes, call the BIOSGuard ACM to do the actual flashing AFAIK

File: verified_boot.md [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+  - Proprietary tooling
+  - Can bypass chipset flash protections
```
Comment (miczyg1) [inline verified_boot.md:222]: Only if such option is selected in MFIT/FITC/FIT on the ME side.

File: verified_boot.md (hunk 4 above)
Comment (miczyg1) [reply verified_boot.md:227]: The available sizes are probably dependent on CPU family and is also configurable in flash descriptor straps IIRC

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: verified_boot.md (hunk 1 above)
Comment (miczyg1) [reply verified_boot.md]: https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR175

File: verified_boot.md (hunk 2 above)
Comment (miczyg1) [reply verified_boot.md]: https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR197

File: verified_boot.md (hunk 3 above)
Comment (miczyg1) [reply verified_boot.md]: Yes, removed: https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR207-R210

File: verified_boot.md (hunk 4 above)
Comment (miczyg1) [reply verified_boot.md]: Confirmed with docs: https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR228-R229

File: verified_boot.md (hunk 5 above)
Comment (miczyg1) [reply verified_boot.md]: https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR47

removed

File: verified_boot.md (hunk 6 above)
Comment (miczyg1) [reply verified_boot.md]: https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR87

File: verified_boot.md (hunk 7 above)
Comment (miczyg1) [reply verified_boot.md]: https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR220-R221

File: verified_boot.md (hunk 8 above)
Comment (miczyg1) [reply verified_boot.md]: https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR222-R223

File: vpub_0xb_verified_boot.md [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+
+- Michał Kopeć (original author)
+  - Firmware Engineer at 3mdeb since 2021
+  - Develops Dasharo for laptops, network appliances and other platforms
+  - Uses Arch btw
+
+- Michał Żygowski (presenter)
+  - Firmware Engineer at 3mdeb since 2017
+  - Develops Dasharo
+  - open-source, HW and security features enthusiats
```
Comment (krystian-hebel) [inline vpub_0xb_verified_boot.md:31]:
```suggestion
  - open-source, HW and security features enthusiast
```

**Review by macpijan — APPROVED** [review]

**Pull request merged**